website/docs/developer/research/research-deep-glacier-migration.md

```diff
@@ -0,0 +1,174 @@
+One option of bulk retrieval is to use python boto3 client for S3:
+
+```python
+import boto3
+
+#this objects list should be provided by the user
+retrieve_objects = ['test1.xml', 'test2.xml']
+
+s3 = boto3.client('s3')
+for retrieve_object in retrieve_objects:
+    try:
+        response = s3.restore_object(
+                Bucket='orca-rhassan-sandbox-glacier', # bucket that contains the objects currently in glacier flexible retrieval
+                Key = retrieve_object,
+                RestoreRequest={
+                    'Days': 30,
+                    'GlacierJobParameters': {
+                        'Tier': 'Bulk',
+                    },
+                },
+            )
+        print(response)
+    except Exception as ex:
+        print(f"{ex} for {retrieve_object}")
+
+```
+Output:
+```
+{'ResponseMetadata': {'RequestId': '61ZWC27NHSFNMYJT', 'HostId': 'ItKZaYvn9rfi6ZVYmUjfZX24mKJlDZlciT9MaKK0aHiNuLWB3Pt8WSEMNt5yBK+5u+MJfzjFDfU=', 'HTTPStatusCode': 202, 'HTTPHeaders': {'x-amz-id-2': 'ItKZaYvn9rfi6ZVYmUjfZX24mKJlDZlciT9MaKK0aHiNuLWB3Pt8WSEMNt5yBK+5u+MJfzjFDfU=', 'x-amz-request-id': '61ZWC27NHSFNMYJT', 'date': 'Sun, 24 Jul 2022 18:01:33 GMT', 'server': 'AmazonS3', 'content-length': '0'}, 'RetryAttempts': 0}}
+{'ResponseMetadata': {'RequestId': 'PNN4XXA5SYTBBRKG', 'HostId': 'HgiIiqfb0hZPHgZml98VCyLLfD5LyQ8pSLpgGN6hwUUMtmcPZoo/ACbCL1rXz+pXZ4Ce2UEe34s=', 'HTTPStatusCode': 202, 'HTTPHeaders': {'x-amz-id-2': 'HgiIiqfb0hZPHgZml98VCyLLfD5LyQ8pSLpgGN6hwUUMtmcPZoo/ACbCL1rXz+pXZ4Ce2UEe34s=', 'x-amz-request-id': 'PNN4XXA5SYTBBRKG', 'date': 'Sun, 24 Jul 2022 18:01:34 GMT', 'server': 'AmazonS3', 'content-length': '0'}, 'RetryAttempts': 0}}
+```
+
+Also see existing ORCA [restore_object function](https://github.com/nasa/cumulus-orca/blob/develop/tasks/request_files/request_files.py#L548)
+Make sure the S3 bucket policy has permission for `s3:PutObject*` action otherwise user will see access denied error. See this [policy](https://github.com/nasa/cumulus-orca/blob/develop/website/docs/developer/deployment-guide/creating-orca-glacier-bucket.md#archive-bucket) for additional details.
+
+Once all files are restored from glacier, there should be some mechanism that will notify users that the retrieval process is complete. One approach could be to enable event notifications in bucket that will trigger either an SQS, SNS or lambda. This will require additional research since AWS notifies for each `s3:ObjectRestore:Completed` API call and not for a specific number of objects.
+
+- After objects are restored, we can use a lambda function or a script that will run after 12 hours and copy the restored objects to deep archive S3 storage using boto3 `copy_object` function as shown below. 
+
+:::important
+Note that there should be some confirmation that all requested objects are successfully restored before running this script.
```

Comment:
How would this confirmation occur? Will this be automated or will an operator need to kickoff this secondary section?

Reply:
better approach would be to kickoff this lambda automatically from s3 event notification once objects are restored.